amr-wind/diffusion/incflo_diffusion.cpp

```diff
@@ -278,6 +283,164 @@ void wall_model_bc(
+
+    amrex::Orientation zlo(amrex::Direction::z, amrex::Orientation::low);
+    amrex::Orientation zhi(amrex::Direction::z, amrex::Orientation::high);
+
+    auto m_store_xy_vel_arr = inst_plane_vel.array();
+    const auto& bxPlanar = inst_plane_vel.box();
+
+    const auto planarlo = amrex::lbound(bxPlanar);
+    const auto planarhi = amrex::ubound(bxPlanar);
+
+    // copies cell center to face
+    Real c0 = 1.0;
+    Real c1 = 0.0;
+
+    // linear extrapolate onto face
+    if(extrapolate)
+    {
+        c0 =  1.5;
+        c1 = -0.5;
+    }
+
+    const Real utau2 = utau*utau;
+
+    for (int lev=0; lev < nlevels; ++lev) {
+        const auto& geom = repo.mesh().Geom(lev);
+        const auto& domain = geom.Domain();
+        MFItInfo mfi_info{};
+
+        auto& rho_lev = density(lev);
+        auto& vel_lev = velocity(lev);
+        auto& eta_lev = viscosity(lev);
+
+        if (Gpu::notInLaunchRegion()) mfi_info.SetDynamic(true);
+#ifdef _OPENMP
+#pragma omp parallel if (Gpu::notInLaunchRegion())
+#endif
+        for (MFIter mfi(vel_lev, mfi_info); mfi.isValid(); ++mfi) {
+            const auto& bx = mfi.validbox();
+            auto vel = vel_lev.array(mfi);
+            auto bc  = vel_lev.array(mfi);
+            auto den = rho_lev.array(mfi);
+            auto eta = eta_lev.array(mfi);
+
+            int idim = 2;
+
+            if (!geom.isPeriodic(idim)) {
+                if (bx.smallEnd(idim) == domain.smallEnd(idim) &&
+                    velocity.bc_type()[zlo] ==  BC::wall_model) {
+                    amrex::ParallelFor(
+                        amrex::bdryLo(bx, idim),
+                        [=] AMREX_GPU_DEVICE(int i, int j, int k) noexcept {
+                            const Real mu  = c0 * eta(i, j, k) + c1 * eta(i, j, k + 1);
+                            // Dirichlet BC
+                            bc(i, j, k - 1, 2) = 0.0;
+                            // Inhomogeneous Neumann BC
+                            bc(i, j, k - 1, 0) = m_store_xy_vel_arr(i, j, planarlo.z, 0)*
+                                den(i, j, k)*utau2/mu;
+                            bc(i, j, k - 1, 1) = m_store_xy_vel_arr(i, j, planarlo.z, 1)*
+                                den(i, j, k)*utau2/mu;
+
+
+                        });
+                }
+
+            }
+
+        } // MFIter loop
+    } // level loop
+}
+
+
+void temp_wall_model_bc(
+    amr_wind::Field& temperature,
+    const amrex::FArrayBox& inst_plane_temp,
+    const amr_wind::FieldState fstate)
+{
+    BL_PROFILE("amr-wind::diffusion::temp_wall_model_bc")
+
+    auto& repo = temperature.repo();
+    auto& density = repo.get_field("density", fstate);
+    auto& alpha = repo.get_field("temperature_mueff");
+    const int nlevels = repo.num_active_levels();
+
+    amrex::Orientation zlo(amrex::Direction::z, amrex::Orientation::low);
+    amrex::Orientation zhi(amrex::Direction::z, amrex::Orientation::high);
+
+    auto m_store_xy_temp_arr = inst_plane_temp.array();
```

Comment:
```suggestion
    auto store_xy_temp_arr = inst_plane_temp.array();
```

`m_` prefix reserved for class/struct instance members